RetailCoder.VBE/Inspections/QuickFixes/PassParameterByReferenceQuickFix.cs

```diff
@@ -12,31 +14,104 @@ namespace Rubberduck.Inspections.QuickFixes
     /// </summary>
     public class PassParameterByReferenceQuickFix : QuickFixBase
     {
-        public PassParameterByReferenceQuickFix(ParserRuleContext context, QualifiedSelection selection)
-            : base(context, selection, InspectionsUI.PassParameterByReferenceQuickFix)
+        private Declaration _target;
+
+        public PassParameterByReferenceQuickFix(Declaration target, QualifiedSelection selection)
+            : base(target.Context, selection, InspectionsUI.PassParameterByReferenceQuickFix)
         {
+            _target = target;
         }
 
         public override void Fix()
         {
-            var parameter = Context.GetText();
+            var argCtxt = GetArgContextForIdentifier(Context, _target.IdentifierName);
 
-            var parts = parameter.Split(new char[]{' '},2);
-            if (1 != parts.GetUpperBound(0))
-            {
-                return;
-            }
-            parts[0] = parts[0].Replace(Tokens.ByVal, Tokens.ByRef);
-            var newContent = parts[0] + " " + parts[1];
+            var terminalNodeImpl = GetByValNodeForArgCtx(argCtxt);
 
-            var selection = Selection.Selection;
+            var replacementLine = GenerateByRefReplacementLine(terminalNodeImpl);
 
-            var module = Selection.QualifiedName.Component.CodeModule;
+            ReplaceModuleLine(terminalNodeImpl.Symbol.Line, replacementLine);
+
+        }
+        private ArgContext GetArgContextForIdentifier(ParserRuleContext context, string identifier)
+        {
+            var procStmtCtx = (ParserRuleContext)context.Parent.Parent;
+            var procStmtCtxChildren = procStmtCtx.children;
+            for (int idx = 0; idx < procStmtCtxChildren.Count; idx++)
+            {
+                if (procStmtCtxChildren[idx] is ArgListContext)
+                {
+                    var argListContext = (ArgListContext)procStmtCtxChildren[idx];
+                    var arg = argListContext.children;
+                    for (int idxArgListCtx = 0; idxArgListCtx < arg.Count; idxArgListCtx++)
+                    {
+                        if (arg[idxArgListCtx] is ArgContext)
+                        {
+                            var name = GetIdentifierNameFor((ArgContext)arg[idxArgListCtx]);
+                            if (name.Equals(identifier))
+                            {
+                                return (ArgContext)arg[idxArgListCtx];
+                            }
+                        }
+                    }
+                }
+            }
+            return null;
```

Comment:
This could be written as a LINQ query, but I'm fine with a `for` loop too.

Instead of `var arg = argListContext.children;`, you could do `var args = argListContext.arg();`, which gives you a `List<ArgContext>`, making the further `if (arg[idxArgListCtx] is ArgContext)` check redundant.

I'd be tempted to simplify it to `var arg = argListContext.arg().SingleOrDefault(parameter => Identifier.GetName(parameter)).Equals(identifier));`